controller/controller.go

```diff
@@ -283,3 +207,21 @@ func convertToEnvVarName(text string) string {
 	}
 	return buffer.String()
 }
+
+func RunKubectlPatch(objectKind string, objectId string, labelName string, labelValue string) {
+	yamlPatch := fmt.Sprintf("spec:\n  template:\n    metadata:\n      labels:\n        %s: '%s'", labelName, labelValue)
+	glog.Infof("About to run patch: %s %s with %s", objectKind, objectId, yamlPatch)
+	attemptDelay := 30
+	for attempt := 1; attempt < 3; attempt++ {
+		err := exec.Command("/kubectl", "patch", objectKind, objectId, "--patch", yamlPatch).Run()
```

Comment:
Is running `kubectl` in a subprocess the only way? Doesn't the Go client support patching operations?

Reply:
Not the only way. The client doesn't do patches well (Update replaces the content, not only patches and this yields to raise conditions). I didn't find any way through the client but any ideas welcome. The only other way I thought about was to use the rest api directly (http requests to the rest api). We could do that too and change the implementation of that function.